extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/HttpClient.java

```diff
@@ -31,6 +37,46 @@
 @NonNullByDefault
 public class HttpClient {
     private int timeout = 1000;
+    private final Logger logger;
+    private final LinkedList<AsyncPutParameters> commandsQueue;
+    private final Runnable commandExecutor;
+    @Nullable
+    private Future<?> job;
+
+    public HttpClient() {
+        logger = LoggerFactory.getLogger(getClass());
+        commandsQueue = new LinkedList<>();
+
+        commandExecutor = () -> {
+            while (true) {
+                try {
+                    int delayTime = 0;
+                    synchronized (this.commandsQueue) {
+                        AsyncPutParameters payloadCallbackPair = HttpClient.this.commandsQueue.poll();
+                        if (payloadCallbackPair != null) {
+                            logger.debug("Async sending put to address: {} delay: {} body: {}",
+                                    payloadCallbackPair.address, payloadCallbackPair.delay, payloadCallbackPair.body);
+                            try {
+                                Result result = HttpClient.this.put(payloadCallbackPair.address,
+                                        payloadCallbackPair.body);
+                                payloadCallbackPair.future.complete(result);
+                            } catch (IOException e) {
+                                payloadCallbackPair.future.completeExceptionally(e);
+                            }
+                            delayTime = payloadCallbackPair.delay;
+                        } else {
+                            return;
+                        }
+                    }
+                    Thread.sleep(delayTime);
+                } catch (InterruptedException e) {
+                    logger.debug("commandExecutorThread was interrupted", e);
```

Comment:
The thread should be marked to be interrupted

Reply:
What exactly do you mean by this? Can you give me a code example?

Reply:
See e.g. https://stackoverflow.com/a/3976377

Another question: Shouldn't we leave the loop if someone interrupt that execution?

Reply:
Good idea, or else we would have an infinite loop. I will make these changes.

Reply:
On second thought I see that this isn't really necessary. If the exception is catched on one message the other messages in the loop should still be sent. And if there are no other messages the loop would be exited on the next run. So this should be okay. I will only make the change for the thread interruption.

Reply:
IMHO the interrupt flag should be set all the time in interrupt exception has been catch, so if you debug it later you can identify the flag.
I don't think this will stop the entire thread.
https://docs.oracle.com/javase/8/docs/api/java/lang/Thread.html#interrupt--

But we can leave it as it is, too.

Reply:
This is the documetion of interrupts from Oracle:

An interrupt is an indication to a thread that it should stop what it is doing and do something else. It's up to the programmer to decide exactly how a thread responds to an interrupt, but it is very common for the thread to terminate. 

I don't know how the ScheduledExecutorService will react to this, so I leave it as it is. Or else I also would have to test the whole binding again for a longer time.